src/utils/spawnAsync.ts

```diff
@@ -54,8 +54,15 @@ export async function spawnAsync(
                 // If cancellation is requested we'll assume that's why it exited
                 return reject(new UserCancelledError());
             } else if (code) {
-                // Replicate the error object of child_process.exec().
-                const error = <ExecError>new Error(localize('vscode-docker.utils.spawn.exited', 'Process exited with code {0}', code));
+                let errorMessage = localize('vscode-docker.utils.spawn.exited', 'Process \'{0}\' exited with code {1}', command.length > 50 ? `${command.substring(0, 50)}...` : command, code);
+
+                if (!onStderr && stderrBuffer) {

```

Comment:
Is there a reason for not just *always* adding the entire error output to the error message, if we have it anyway?

Reply:
The main reason was the tasks. For them, `onStderr` already results in the standard error output being emitted to the task pseudoterminal window in nice red font. If we put it in the error message as well in this scenario, the same stderr output gets re-emitted to the terminal by https://github.com/microsoft/vscode-docker/blob/master/src/tasks/DockerTaskProvider.ts#L56, which is redundant.

Reply:
Generally speaking, the assumption is that if the caller specifies `onStderr`, they are handling error output themselves and putting it in the error message would be redundant; whereas if `onStderr` is `undefined`, then the output would be lost entirely if we _don't_ put it in the error message.